src/unix/core.c

```diff
@@ -983,3 +984,61 @@ int uv__dup2_cloexec(int oldfd, int newfd) {
     return r;
   }
 }
+
+
+int uv_os_homedir(char* buffer, size_t* size) {
+  struct passwd pw;
+  struct passwd* result;
+  char* buf;
+  uid_t uid;
+  size_t bufsize;
+  size_t len;
+  int r;
+
+  if (buffer == NULL || size == NULL || *size == 0)
+    return -EINVAL;
+
+  bufsize = sysconf(_SC_GETPW_R_SIZE_MAX);
+
+  if (bufsize <= 0)
+    return -EIO;
+
+  uid = getuid();
```

Comment:
I commented on that [here](https://github.com/libuv/libuv/pull/350#discussion_r30469592):

> As to whether uid should be the result of getuid() or geteuid(), I _think_ getuid() is proper (it's what e.g. Ruby's Etc module defaults to), although I suppose it may be somewhat surprising for suid root binaries.

I'm 75% sure python's `os.path.expanduser()` works the same way when HOME is unset.  I'm inclined to follow the behavior of other projects.